tests/unit/string_lib_test.c

```diff
@@ -496,6 +496,42 @@ static void test_string_to_long_compatibility(void)
     // Old function (StringToLongUnsafe) does not allow trailing whitespace
 }
 
+static void test_string_decimal_to_long(void)
+{
+   // Basic usage:
+   assert_int_equal(StringDecimalToLong("0.0"), 0);
+   assert_int_equal(StringDecimalToLong("-0.0"), 0);
+   assert_int_equal(StringDecimalToLong("+0.0"), 0);
+   assert_int_equal(StringDecimalToLong("123.0"), 123);
```

Comment:
`StringToLong()` and `StringDecimalToLong()` return the error code (0), not the result. All the tests are failing because of this. You need to use the output parameter, and check it's value after each call.

```C
   assert_int_equal(StringDecimalToLong("123.0", &value_out), 0);
   assert_int_equal(value_out, 123);
```

(To avoid a lot of repetition, you could put this in a wrapper function, or macro).